js/private/js_image_layer.mjs

```diff
@@ -1,9877 +1,339 @@
-                    if (prev !== null) {
-                        prev.next = node.next;
-                    } else if (node.next !== null) {
-                        listeners.set(eventName, node.next);
-                    } else {
-                        listeners.delete(eventName);
-                    }
-                } else {
-                    prev = node;
-                }
-
-                node = node.next;
-            }
+function _mtree_link_line(key, linkname) {
+    const link_parent = path.dirname(key)
+    linkname = path.relative(link_parent, linkname)
 
-            // Add new value.
-            if (listener !== null) {
-                const newNode = {
-                    listener,
-                    listenerType: ATTRIBUTE,
-                    passive: false,
-                    once: false,
-                    next: null,
-                };
-                if (prev === null) {
-                    listeners.set(eventName, newNode);
-                } else {
-                    prev.next = newNode;
-                }
-            }
-        },
-        configurable: true,
-        enumerable: true,
-    }
+    // interestingly, bazel 5 and 6 sets different mode bits on symlinks.
+    // well use `0o755` to allow owner&group to `rwx` and others `rx`
+    // see: https://chmodcommand.com/chmod-775/
+    return `${vis(key)} uid={{UID}} gid={{GID}} time=0 mode=0775 type=link link=${vis(linkname)}`;
 }
 
-/**
- * Define an event attribute (e.g. `eventTarget.onclick`).
- * @param {Object} eventTargetPrototype The event target prototype to define an event attrbite.
- * @param {string} eventName The event name to define.
- * @returns {void}
- */
-function defineEventAttribute(eventTargetPrototype, eventName) {
-    Object.defineProperty(
-        eventTargetPrototype,
-        `on${eventName}`,
-        defineEventAttributeDescriptor(eventName)
-    );
+function _mtree_file_line(key, content) {
+    const dest = vis(key)
+    // Due to filesystems setting different bits depending on the os we have to opt-in
+    // to use a stable mode for files.
+    // In the future, we might want to hand off fine-grained control of these to users
+    // see: https://chmodcommand.com/chmod-0555/
+    return `${dest} uid={{UID}} gid={{GID}} time=0 mode=0555 type=file content=${vis(content)}`;
 }
 
-/**
- * Define a custom EventTarget with event attributes.
- * @param {string[]} eventNames Event names for event attributes.
- * @returns {EventTarget} The custom EventTarget.
- * @private
- */
-function defineCustomEventTarget(eventNames) {
-    /** CustomEventTarget */
-    function CustomEventTarget() {
-        EventTarget.call(this);
-    }
 
-    CustomEventTarget.prototype = Object.create(EventTarget.prototype, {
-        constructor: {
-            value: CustomEventTarget,
-            configurable: true,
-            writable: true,
-        },
-    });
+async function split() {	
+    const UID = "{{UID}}"
```

Comment:
I think these are unused because `{{UID}}` is used everywhere, so it's always inlined...